compiler/src/dotty/tools/dotc/transform/LinkScala2Impls.scala

```diff
@@ -75,7 +75,7 @@ class LinkScala2Impls extends MiniPhase with IdentityDenotTransformer { thisPhas
     def currentClass = ctx.owner.enclosingClass.asClass
     app match {
       case Apply(sel @ Select(Super(_, _), _), args)
-      if sel.symbol.owner.is(Scala2x) && currentClass.mixins.contains(sel.symbol.owner) =>
+      if sel.symbol.owner.is(Scala2x) && currentClass.mixins.contains(sel.symbol.owner) && !ctx.settings.scalajs.value =>
```

Comment:
I think we could go further and use the default method even when not using Scala.js, going through the the static methods is supposed to improve cold performance but based on the investigation I did, they shouldn't affect semantics (https://github.com/lampepfl/dotty/issues/5928)

Reply:
I don't object, but I would argue that it is out of scope of this PR. I would keep this PR to be Scala.js-only. Changing the JVM behavior should be done in a separate PR I think.

Reply:
Your call, but I'd like to keep the difference in code paths taken when scala.js is enabled to a minimum.

Reply:
I haven't managed to make it work for the JVM in 20 minutes. I think changing the JVM should be best done by someone who understand the thing better than I do.